src/Swagger/Extractor/CollectionGetExtractor.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Swagger\Extractor;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use ApiPlatform\Core\Swagger\Model\SwaggerDocumentation;
+use ApiPlatform\Core\Swagger\Model\SwaggerMetadata;
+use ApiPlatform\Core\Swagger\Util\SwaggerDefinitions;
+use ApiPlatform\Core\Swagger\Util\SwaggerFilterDefinitions;
+use Symfony\Component\HttpFoundation\Request;
+
+class CollectionGetExtractor implements SwaggerExtractorInterface
+{
+    /**
+     * @var SwaggerExtractorInterface
+     */
+    private $decoratedExtractor;
+
+    /**
+     * @var ResourceMetadataFactoryInterface
+     */
+    private $resourceMetadataFactory;
+
+    /**
+     * @var SwaggerDefinitions
+     */
+    private $swaggerDefinitions;
+
+    /**
+     * @var SwaggerFilterDefinitions
+     */
+    private $swaggerFilterDefinitions;
+
+    /**
+     * RouteDocumentationExtractor constructor.
+     *
+     * @param ResourceMetadataFactoryInterface $resourceMetadataFactory
+     * @param SwaggerDefinitions               $swaggerDefinitions
+     * @param SwaggerFilterDefinitions         $swaggerFilterDefinitions
+     * @param SwaggerExtractorInterface        $decoratedExtractor
+     */
+    public function __construct(
+        ResourceMetadataFactoryInterface $resourceMetadataFactory,
+        SwaggerDefinitions $swaggerDefinitions,
+        SwaggerFilterDefinitions $swaggerFilterDefinitions,
+        SwaggerExtractorInterface $decoratedExtractor = null
+    ) {
+        $this->decoratedExtractor = $decoratedExtractor;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->swaggerDefinitions = $swaggerDefinitions;
+        $this->swaggerFilterDefinitions = $swaggerFilterDefinitions;
+    }
+
+    public function extract(SwaggerMetadata $swaggerMetadata): SwaggerDocumentation
+    {
+        if (null !== $this->decoratedExtractor) {
+            $documentation = $this->decoratedExtractor->extract($swaggerMetadata);
+        } else {
+            $documentation = new SwaggerDocumentation();
+        }
+
+        if ($swaggerMetadata->isCollection() && in_array(Request::METHOD_GET, $swaggerMetadata->getMethods())) {
```

Comment:
Please use the string `GET` directly to not introduce a dependency to Symfony's HttpFoundation for this component.